pandas/io/excel/_base.py

```diff
@@ -33,7 +33,7 @@
 
 Parameters
 ----------
-io : str, file descriptor, pathlib.Path, ExcelFile or xlrd.Book
+io : str, ExcelFile, xlrd.Book, path object, URL or file-like object
```

Comment:
strictly speaking, in practice the URL is also a str or path object, so not sure we should list it in the types. It is clearly mentioned in the description below.

Reply:
addressed in 4eb3fa5

Reply:
I did the same in 6a3704bccbcb34e51513c2d0836b028ef5f2f4bc  for all other read_* methods I touched